### PR TITLE
Add methods for including external source GUID and Name

### PR DESCRIPTION
**Pull request by cong78**
Tests are are not changed. Let me know how can we work on it.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -144,7 +144,7 @@
         <spring-security.version>5.1.5.RELEASE</spring-security.version>
         <tomcat.version>9.0.21</tomcat.version>
         <commons-io.version>2.4</commons-io.version>
-        <commons-collections.version>4.4</commons-collections.version>
+        <commons-collections.version>4.</commons-collections.version>
```
Comment (popa-raluca) [inline pom.xml]: did you remove this on purpose? the build will fail with "4." version

File: open-metadata-implementation/common-services/ocf-metadata-management/ocf-metadata-handlers/src/main/java/org/odpi/openmetadata/commonservices/ocf/metadatamanagement/handlers/SchemaTypeHandler.java [hunk 2]
```diff
@@ -480,6 +480,58 @@ public void  saveSchemaAttributes(String                userId,
...
+        {
+            for (SchemaAttribute  schemaAttribute : schemaAttributes)
+            {
+                if (schemaAttribute != null)
+                {
+                    String schemaAttributeGUID = this.findSchemaAttribute(userId, schemaAttribute, methodName);
+
+                    if (schemaAttributeGUID == null)
+                    {
+                        schemaAttributeGUID = addSchemaAttribute(userId, schemaAttribute);
```
Comment (popa-raluca) [inline open-metadata-implementation/common-services/ocf-metadata-management/ocf-metadata-handlers/src/main/java/org/odpi/openmetadata/commonservices/ocf/metadatamanagement/handlers/SchemaTypeHandler.java]: you should use addExternalSchemaAttribute here

File: open-metadata-implementation/common-services/repository-handler/src/main/java/org/odpi/openmetadata/commonservices/repositoryhandler/RepositoryHandler.java [hunk 3]
```diff
@@ -2097,6 +2266,51 @@ public void createRelationship(String                  userId,
...
+                                           String                  relationshipTypeGUID,
+                                           String                  end1GUID,
+                                           String                  end2GUID,
+                                           InstanceProperties      relationshipProperties,
+                                           String                  methodName) throws UserNotAuthorizedException,
+            PropertyServerException
+    {
+        try
+        {
+            metadataCollection.addExternalRelationship(userId,
```
Comment (popa-raluca) [inline open-metadata-implementation/common-services/repository-handler/src/main/java/org/odpi/openmetadata/commonservices/repositoryhandler/RepositoryHandler.java:2296]: Wrong order for the parameters when calling addExternal. The signature for addExternalRelationship is  public Relationship addExternalRelationship(String               userId,
                                                String               relationshipTypeGUID,
                                                String               externalSourceGUID,
                                                String               externalSourceName,
                                                InstanceProperties   initialProperties,
                                                String               entityOneGUID,
                                                String               entityTwoGUID,
                                                InstanceStatus       initialStatus)

**Review by popa-raluca — CHANGES_REQUESTED** [review]
Looks good so far, appart from the few comments. I wouldn't merge it to data-engine branch until the tests are updated, because otherwise it will not compile anymore.
Also, not sure if you've done some dev tests with the initial_load payload, maybe I can help with that once it is ready to be merged into data-engine branch.

**Comment by cong78** [comment]
> Looks good so far, appart from the few comments. I wouldn't merge it to data-engine branch until the tests are updated, because otherwise it will not compile anymore.
> Also, not sure if you've done some dev tests with the initial_load payload, maybe I can help with that once it is ready to be merged into data-engine branch.

Cool thanks. I will make the changes today.

File: pom.xml (hunk 1 above)
Comment (cong78) [reply pom.xml]: Merge mistake I think

**Review by popa-raluca — APPROVED** [review]

**Pull request merged**